### ISSUE TITLE
schema loading from apollo engine schema registry

### ISSUE THREAD

**Issue by danilobuerger** [dotansimha/graphql-code-generator#3047]
**Is your feature request related to a problem? Please describe.**

Apollo Engine provides a Schema Registry:

> Apollo Graph Manager provides a schema registry that serves as a central hub for managing your data graph (as recommended by Principled GraphQL). The schema registry enables you to explore your schema's types and fields, track its change history, and improve your production server's security by removing the need for introspection. You can also register variants of your schema to support environments besides production, such as test or staging.

https://www.apollographql.com/docs/graph-manager/schema-registry/

It would be nice if the schema could be loaded from there.

**Describe the solution you'd like**

The codegen config could look like:
```
schema: engine#id/tag
generates:
  ./src/types.ts:
    plugins:
      - typescript
```
which would download from engine the schema from the graph named `id` at variant `tag`.

The required engine api key could come as an environment variable.

**Describe alternatives you've considered**

This can already be achieved using a custom loader (https://graphql-code-generator.com/docs/getting-started/schema-field#custom-schema-loader) but it would be nice to have this directly integrated.

**Additional context**

The implementation for apollo cli can be found here:
https://github.com/apollographql/apollo-tooling/blob/master/packages/apollo-language-server/src/providers/schema/engine.ts

**Comment by dotansimha** [comment]
@kamilkisiela we can create a new loader in graphql-toolkit for this?

**Comment by ardatan** [comment]
@danilobuerger Could you try with the following version?
`1.9.2-alpha-6a0f647a.11+6a0f647a`

The configuration should be like below;
```yml
schema:
   - apollo-engine:
      engine:
         apiKey: [API-KEY]
      client:
         service: [ID]@[TAG]
```

**Comment by danilobuerger** [comment]
@ardatan Sure will give it a try and report back.

However, I think it's a really bad idea to put the api key in the codegen config. People will check it in and then leak it everywhere. This should really be an environment variable.

**Comment by ardatan** [comment]
@danilobuerger You can pass it as environmental variable;
```yml
schema:
   - apollo-engine:
      engine:
         apiKey: ${API_KEY}
      client:
         service: [ID]@[TAG]
```
So API_KEY would be coming from env variable.

**Comment by danilobuerger** [comment]
I didn't know that that was possible, thanks for pointing it out! For future reference, its also documented here: https://graphql-code-generator.com/docs/getting-started/codegen-config#environment-variables

**Comment by danilobuerger** [comment]
@ardatan Yup, works on my machine. Thanks a lot for all the hard work!

**Comment by ardatan** [comment]
@danilobuerger  Great! Thank you for pointing out the code in the apollo-language-server. So it helped a lot while implementing this feature :)

**Comment by dotansimha** [comment]
Available in `v1.10.0`.
